src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -122,8 +123,8 @@ public function send(string $body, array $headers, int $delay = 0): string
             ':body' => $body,
             ':headers' => json_encode($headers),
             ':queue_name' => $this->configuration['queue_name'],
-            ':created_at' => self::formatDateTime($now),
-            ':available_at' => self::formatDateTime($availableAt),
+            ':created_at' => $this->convertDateTime($now),
+            ':available_at' => $this->convertDateTime($availableAt),
```

Comment:
We could even use the datetime directly as parameter and pass the Doctrine DBAL types (in the third argument) for these, so that DBAL handles the conversion calls for us (and would avoid the need to create a public `convertDateTime` too, as tests can do the same)

Reply:
Sorry I don't understand.  
Because in this file, the `executeQuery` method is not the same method of https://github.com/doctrine/dbal/blob/66f28111d57fdc3508d4c97989257291702176cf/lib/Doctrine/DBAL/Connection.php#L883.

Reply:
it is the same. The `driverConnection` propert contains a `Doctrine\DBAL\Connection`, which is exactly the code I linked.

Reply:
I think @stof means we could use `\Doctrine\DBAL\Connection::executeQuery` which has the types as parameter.